encfs/compatwin.cpp

```diff
@@ -640,22 +640,44 @@ unix::stat(const char *path, struct stat_st *buffer)
     return -1;
   }
 
-
   // We need an active file handle in order to get the file index ID 
   HANDLE hff = CreateFileW(fn.c_str(), GENERIC_READ,
     FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE,
     NULL, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, NULL);
-  if (hff == INVALID_HANDLE_VALUE) {
-    errno = ERRNO_FROM_WIN32(GetLastError());
-    return -1;
-  }
-  BY_HANDLE_FILE_INFORMATION wfd;
-  if (!GetFileInformationByHandle(hff, &wfd)) {
-    errno = ERRNO_FROM_WIN32(GetLastError());
-    return -1;
-  }
-  CloseHandle(hff);
 
+  BY_HANDLE_FILE_INFORMATION hfi;
+  WIN32_FIND_DATAW wfd;
+
+  // Not sure about the default values after init, so in doubt...
+  hfi.dwFileAttributes = 0;
```

Comment:
Should we just memset (or zero-init) these two structs and get rid of these init lines?  

Also, what do you think about instead creating DWORDs here to store these values like is done with the file time values, then we don't have to add the values together between the two return structures (like with `(hfi.dwFileAttributes + wfd.dwFileAttributes)`, etc.)  Would this be more confusing or less confusing?

Reply:
@jetwhiz, up to you !
If you find a better coding style, of course feel free !
I agree that having to sum the attributes from the 2 structures is not so easy to read.
Goal was just to avoid having the same twice (for `GetFileInformationByHandle` and for `FindFirstFileW`).